File: bindings/java/src/main/java/org/indunet/robot/bus/TimerHandle.java
package org.indunet.robot.bus;

import com.sun.jna.Pointer;

/** Opaque timer handle returned by {@link Node#createTimer}. */
public final class TimerHandle implements AutoCloseable {
    final Pointer ptr;

    TimerHandle(Pointer ptr) {
        this.ptr = ptr;
    }

    @Override
    public void close() {
        RobotBusC.Holder.INSTANCE.robot_bus_timer_handle_free(ptr);
    }
}

File: bindings/android/src/main/kotlin/org/indunet/robot/bus/TimerHandle.kt
package org.indunet.robot.bus

import com.sun.jna.Pointer

/** Opaque timer handle returned by [Node.createTimer]. */
class TimerHandle internal constructor(internal val ptr: Pointer?) : AutoCloseable {
    override fun close() {
        RobotBusC.Holder.INSTANCE.robot_bus_timer_handle_free(ptr)
    }
}
